Source/ISeriesProvider/DB2iSeriesSqlBuilder.cs

```diff
@@ -139,7 +133,80 @@ public override StringBuilder BuildTableName(StringBuilder sb, string server, st
 			if (database != null && schema == null)
 				throw new LinqToDBException($"{Provider.Name} requires schema name if database name provided.");
 
-			return base.BuildTableName(sb, null, database, schema, table);
+			return base.BuildTableName(sb, null, database, schema, table, tableOptions);
+		}
+
+		protected override void BuildCreateTableCommand(SqlTable table)
+		{
+			string command;
+
+			if (table.TableOptions.IsTemporaryOptionSet())
+			{
+				switch (table.TableOptions & TableOptions.IsTemporaryOptionSet)
+				{
+					case TableOptions.IsTemporary:
+					case TableOptions.IsTemporary | TableOptions.IsLocalTemporaryData:
+					case TableOptions.IsTemporary | TableOptions.IsLocalTemporaryStructure:
+					case TableOptions.IsTemporary | TableOptions.IsLocalTemporaryStructure | TableOptions.IsLocalTemporaryData:
+					case TableOptions.IsLocalTemporaryData:
+					case TableOptions.IsLocalTemporaryStructure:
+					case TableOptions.IsLocalTemporaryStructure | TableOptions.IsLocalTemporaryData:
+						command = "DECLARE GLOBAL TEMPORARY TABLE ";
+						break;
+					case TableOptions.IsGlobalTemporaryStructure:
+					case TableOptions.IsGlobalTemporaryStructure | TableOptions.IsLocalTemporaryData:
+						command = "CREATE GLOBAL TEMPORARY TABLE ";
+						break;
+					case var value:
+						throw new InvalidOperationException($"Incompatible table options '{value}'");
+				}
+			}
+			else
+			{
+				command = "CREATE TABLE ";
+			}
+
+			StringBuilder.Append(command);
+		}
+
+		protected override void BuildStartCreateTableStatement(SqlCreateTableStatement createTable)
```

Comment:
I'm not sure this will work on DB2i . The tests at Tets.Linq\TableOptionTests should test against this.